src/components/main/LineBlock.tsx

```diff
@@ -0,0 +1,147 @@
+import { EditCtx } from '../../store/context_stores/Contexts';
+import { useValues } from '../../store/StateHooks';
+import { RemappedLine, Modification, LineMenuItem } from '../../../types/types';
+
+
+const LineContext = createContext( null );
+
+export { LineContext };
+
+interface NewProps {
+  line: RemappedLine,
+  lineMod: Modification,
+  entryID: string
+  selectedElement?: string,
+  isMainLine?: any
+}
+
+
+const GurmukhiMenu: LineMenuItem[] = [
+  {
+    title: 'Copy Unicode',
+    action: ( ascii ) => { Clipboard.setString( unicode( ascii ) ); },
+  },
+  {
+    title: 'Copy Ascii',
+    action: ( ascii ) => { Clipboard.setString( ascii ); },
+  },
+];
+
+const RomanMenu: LineMenuItem[] = [
+  {
+    title: 'Copy (Default)',
+    action: ( ascii ) => { Clipboard.setString( ascii ); },
+  },
+];
+
+const LineBlock: React.FC<NewProps> = ( { line, isMainLine } ) => {
+  const { fontSizes, displayElements, sources } = useValues( 'viewerModel' );
+  const isEditMode = EditCtx.useStoreState( ( state ) => state.isEditMode );
+
+  const {
+    gurmukhi, eng, teeka, translit,
+  } = fontSizes;
+  const {
+    displayEng, displayTeeka, displayTranslit, displayVishraams,
+  } = displayElements;
+
+  const {
+    Gurbani, Translations, Transliteration, id, Vishraams,
+  } = line;
+
+  const { ascii } = Gurbani;
+  const Pangtee = (
+    <TextBlockBase
+      isSelected={false}
+      isMainLine={isMainLine}
+      lineID={id}
+    >
+      <GurmukhiTextContainer style={{ fontSize: gurmukhi }}>
+        <VishraamText
+          vishraams={displayVishraams ? Vishraams : {}}
+          source={sources.vishraamSource}
+          lineID={id}
+        />
+      </GurmukhiTextContainer>
+    </TextBlockBase>
+  );
+
+
+  const Transl = (
+    <TextBlockBase
+      isSelected={false}
+      lineID={id}
+    >
+      <RomanTextContainer style={{ fontSize: eng }}><BaseText /></RomanTextContainer>
+    </TextBlockBase>
+  );
+
+  const Teeka = (
+    <TextBlockBase
+      isSelected={false}
+      lineID={id}
+    >
+      <GurmukhiTextContainer style={{ fontSize: teeka }}><BaseText /></GurmukhiTextContainer>
+    </TextBlockBase>
+  );
+  const Translit = (
+    <TextBlockBase
+      isSelected={false}
+      lineID={id}
+    >
+      <RomanTextContainer style={{ fontSize: translit }}>
+        <VishraamText
+          vishraams={displayVishraams ? Vishraams : {}}
+          source={sources.vishraamSource}
+          lineID={id}
+        />
+      </RomanTextContainer>
+    </TextBlockBase>
+  );
+
+  const TextNodes: any[][] = [
+    [ Pangtee, [ true, ascii, GurmukhiMenu ] ],
+    [ Transl, [ displayEng, Translations[sources.translationLang], RomanMenu ] ],
+    [ Teeka, [ displayTeeka, Translations.Punjabi[sources.teekaSource], GurmukhiMenu ] ],
+    [ Translit, [ displayTranslit, Transliteration[sources.translitLang], RomanMenu ] ],
+  ];
+  return (
+    <View style={style.column} key={`LineBlock-${id}-${ascii}`}>
```

Comment:
from the react docs: `The key only has to be unique among its siblings, not globally unique.` https://reactjs.org/docs/reconciliation.html#keys